com.microsoft.mrtk.input/Visualizers/ControllerVisualizer/ControllerVisualizer.cs

```diff
@@ -89,6 +89,11 @@ private void RenderControllerVisuals(InputAction.CallbackContext context)
             RenderControllerVisuals(context.control.device);
         }
 
+        /// <summary>
+        /// A cache used internally to store the platform visuals that are used for an input device
+        /// </summary>
+        private static Dictionary<UnityInputSystem.InputDevice, (bool, GameObject)> inputDevicePlatformVisualsCache = new Dictionary<UnityInputSystem.InputDevice, (bool, GameObject)>();
```

Comment:
This feels a _little_ weird, since we now have two caches with the same `GameObject`s stored in them. Is there anyway we could rework this (on either side) to dedupe the caches? Keeping two caches in sync is always a pain!

Reply:
Yeah it's not great! Do you think that the input device should ultimately be the "cached source" vs. a model key? Are there cases where multiple devices might use the same key? If not, we can move everything onto the ModelLoader side of things.

Reply:
Is this discussion still active @RogPodge @keveleigh ?

Reply:
I think I fixed it in the latest commits

Reply:
Oooh, I _don't_ think we should use the input device as the cached source.
The main scenario that would break would be a user disconnecting one model of controller (say, the original WMR ones)
![image](https://user-images.githubusercontent.com/3580640/193930009-b78c1c7a-f851-41d6-9d4b-fedeaa8112c3.png)
and connecting the Samsung ones
![image](https://user-images.githubusercontent.com/3580640/193930056-8e78bfa8-dbba-4ac0-a68e-786f9c6eb700.png)
in most cases, this will use the same `InputDevice` on the Unity side, I believe, since they represent the same set of interactions, but have different models (and thus model keys).

So, I wouldn't expect multiple devices to use the same key, but I would expect one device to possibly have multiple keys.

Reply:
Going to re-address this whole concept by just having a bool which prevents the error from being fired consecutively, given the inability for us to truly differentiate controllers before attempting to fetch the model key.

Reply:
All fixed up now!

Reply:
Is there still some "work" being done every time a non-controller-model device is detected?

Reply:
According to Kurtis lookup of the model key is pretty low cost and we do have to query for it every time if we don't want to mis-represent a controller. Because input devices are reused for different controllers, we have to cache based on model key to give an accurate visualization.